### PR TITLE
[Feat] Adds discrete gaussian from project-monai

### LINKED ISSUE #559: [Feat] Discrete Gaussian kernel (gaussian) should be adapted for discrete filtering
## 🐛 Bug
The [current Gaussian kernel](https://github.com/kornia/kornia/blob/9a49b3c473fbbef6df1e0f245f8e99e36f6fec6a/kornia/filters/kernels.py#L19) simply samples the continuous Gaussian function:
```python
def gaussian(window_size, sigma):
    x = torch.arange(window_size).float() - window_size // 2
    if window_size % 2 == 0:
        x = x + 0.5
    gauss = torch.exp((-x.pow(2.0) / float(2 * sigma ** 2)))
    return gauss / gauss.sum()
```

## Expected behavior
Discrete Gaussian filtering should be compatible with scale-space theory and rely on modified Bessel function for its kernel weights. See e.g.:
https://en.wikipedia.org/wiki/Scale_space_implementation#The_discrete_Gaussian_kernel

## Additional context
Note that this is a similar issue to that raised in MONAI:
https://github.com/Project-MONAI/MONAI/issues/363

I came here while looking for a PyTorch implementation of a Discrete Gaussian filter but realised this one was also sampling the continuous Gaussian function.

ITK implements such a behaviour in itk::GaussianOperator:
https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Core/Common/include/itkGaussianOperator.hxx

[Wikipedia](https://commons.wikimedia.org/wiki/File:Discrete_Gaussian_kernel.svg) also provides a basic python implementation (albeit relying on scipy and without normalisation):
```python
from scipy.special import iv
import numpy as np

def discrete_gaussian_kernel(n, t):
    T = np.exp(-t) * iv(n, t)
    return T
```

### PR DESCRIPTION
**Pull request by wyli**
### Description
- fixes #559 
- minor docstring fixes in kornia/filters/gaussian.py

see also 
- https://github.com/Project-MONAI/MONAI/issues/363
- https://github.com/Project-MONAI/MONAI/pull/1097
- https://github.com/Project-MONAI/MONAI/pull/1020

(pls feel free to revise this branch directly)

### Status
**ready**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New tests added to cover the changes
- [x] Docstrings/Documentation updated

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our 
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [x] Did you discuss the functionality or any breaking changes before ?
- [x] **Pass all tests**: did you test in local ? `make test`
- [x] Unittests: did you add tests for your new functionality ?
- [x] Documentations: did you build documentation ? `make build-docs`
- [x] Implementation: is your code well commented and follow conventions ? `make lint`
- [x] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [ ] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>
  
  - [ ] Assign correct label
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if 
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?
 
</details>

### CODE REVIEW

File: kornia/filters/kernels.py [hunk 1]
```diff
@@ -24,6 +24,98 @@ def gaussian(window_size, sigma):
     return gauss / gauss.sum()
 
 
+def gaussian_discrete_erf(window_size, sigma):
+    r"""discrete Gaussian by interpolating the error function.
+    adapted from:
+        https://github.com/Project-MONAI/MONAI/blob/master/monai/networks/layers/convutils.py
+    """
+    sigma = torch.as_tensor(sigma, dtype=torch.float, device=sigma.device if torch.is_tensor(sigma) else None)
```
Comment (edgarriba) [inline kornia/filters/kernels.py:32]: `torch.is_tensor` it's not supported with torchscript - I would rather use `isinstance(sigma, torch.Tensor)` as we have been using across the library and it's more pythonic :)

File: kornia/filters/kernels.py [hunk 2]
```diff
@@ -24,6 +24,98 @@ def gaussian(window_size, sigma):
...
+    x = torch.arange(window_size).float() - window_size // 2
+    t = 0.70710678 / torch.abs(sigma)
+    gauss = 0.5 * ((t * (x + 0.5)).erf() - (t * (x - 0.5)).erf())
+    gauss = gauss.clamp(min=0)
+    return gauss / gauss.sum()
+
+
+def _modified_bessel_0(x: torch.Tensor) -> torch.Tensor:
+    r"""adapted from:
+        https://github.com/Project-MONAI/MONAI/blob/master/monai/networks/layers/convutils.py
```
Comment (edgarriba) [inline kornia/filters/kernels.py:42]: :+1:

**Review by edgarriba — APPROVED** [review]
LGTM. Can you add to docs here ? https://github.com/kornia/kornia/blob/master/docs/source/filters.rst#kernels

**Review by edgarriba — APPROVED** [review]
ready to go
